cpp/include/raft/spatial/knn/detail/ball_cover.cuh

```diff
@@ -64,9 +65,12 @@ namespace detail {
  * @param handle
  * @param index
  */
-template <typename value_idx, typename value_t, typename value_int = std::uint32_t>
+template <typename value_idx,
+          typename value_t,
+          typename value_int  = std::uint32_t,
+          typename matrix_idx = std::uint32_t>
```

Comment:
Here too wrt new template args. If we're defaulting it to be the same as `value_int`, can we just reuse it?